26). Add topic files 75 to 78 (#18050)
dotnet/docs: docs/standard/linq/list-all-nodes-tree.md

---
title: How to list all nodes in a tree - LINQ to XML
description: You can use XPath to list all nodes in an XML tree. This article provides an example for C# and Visual Basic that lists the nodes of a tree. Each node is represented by an XPath expression that specifies the location of the node in the tree.
ms.date: 07/20/2015
dev_langs:
  - "csharp"
  - "vb"
ms.assetid: 3e934371-f4c6-458b-9f6b-f9061b596f5b
---

# How to list all nodes in a tree (LINQ to XML)

> Sometimes it is helpful to list all nodes in a tree – for example, to learn exactly how a method or property affects the tree. The example below for C# and Visual Basic lists the nodes of a tree. Each node is represented by an XPath expression that specifies the location of the node in the tree.

Reviewer: ```suggestion
Sometimes, it's helpful to list all nodes in a tree – for example, to learn exactly how a method or property affects the tree. The example in this article lists the nodes of a tree in C# or Visual Basic. Each node is represented by an XPath expression that specifies the location of the node in the tree.
```

            XElement el = xobj as XElement;
            if (el != null)
                return "/" + NameWithPredicate(el);

>             // the XPath data model doesn't include white space text nodes

Reviewer: ```suggestion
            // The XPath data model doesn't include white space text nodes
```
